File: java/QuickSort.java
public class QuickSort{
    public static void main(String[] args) {
        int[] array = {8, 2, 3, 4, 7, 9, 1, 5, 6};

        sort(array, 0, array.length - 1);

        for(int i: array){
            System.out.print(i + " ");
        }
    }

    private static void sort(int[] array, int start, int end){
        if(end<= start) return;
        int pivot = partition(array, start, end);

        sort(array, start, pivot - 1);
        sort(array, pivot + 1, end);
    }

    private static int partition(int[] array, int start, int end){
        int pivot = array[end];
        int i = start-1;

        for(int j=start; j<=end - 1; j++){
            if(array[j] < pivot){
                i++;
                int temp = array[i];
                array[i] = array[j];
                array[j] = temp;
            }
        }
        i++;
        int temp = array[i];
        array[i] = array[end];
        array[end] = temp;
        return i;
    }
}

File: python/quick_sort.py

def sort(arr, start, end):
    if end<=start: 
        return
    pivot = partition(arr, start, end)

    sort(arr, start, pivot-1)
    sort(arr, pivot+1, end) 


def partition(arr, start, end):
    pivot = arr[end]
    i = start - 1

    for j in range(end-1):
        if arr[j] < pivot:
            i+=1
            tmp = arr[i]
            arr[i] = arr[j]
            arr[j] = tmp
        j+=1
    i+=1
    tmp = arr[i]
    arr[i] = arr[end]
    arr[end] = tmp
    return i

arr = [4, 7, 9, 3, 2, 5]
print(sort(arr, 0, len(arr)-1))